app/V1Module/presenters/AssignmentSolutionReviewsPresenter.php

```diff
@@ -0,0 +1,407 @@
+
+        // get and verify inputs
+        $req = $this->getRequest();
+        $text = trim($req->getPost("text"));
+        if (!$text) {
+            throw new BadRequestException("The text of the comment must not be empty.");
+        }
+
+        $issue = $req->getPost("issue") !== null ? filter_var($req->getPost("issue"), FILTER_VALIDATE_BOOLEAN) : null;
+        $issueChanged = $issue !== null && $comment->isIssue() !== $issue;
+
+        if ($text !== $comment->getText() || $issueChanged) {
+            // modification needed
+            $oldText = $comment->getText();
+            $comment->setText($text);
+            if ($issue !== null) {
+                $comment->setIssue($issue);
+            }
+            $this->reviewComments->persist($comment);
+
+            if ($solution->getReviewedAt() !== null) {
+                // review is already closed, this needs special treatement
+                if ($issueChanged) {
+                    $solution->setIssuesCount($solution->getIssuesCount() + ($issue ? 1 : -1));
+                    $this->assignmentSolutions->persist($solution);
+                }
+
+                $suppressNotification = filter_var($req->getPost("suppressNotification"), FILTER_VALIDATE_BOOLEAN);
+                if (!$suppressNotification) {
+                    $this->reviewsEmailSender->changedReviewComment($solution, $comment, $oldText, $issueChanged);
+                }
+            }
+        }
+
+        $this->sendSuccessResponse($comment);
+    }
+
+    public function checkDeleteComment(string $id, string $commentId)
+    {
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        $comment = $this->reviewComments->findOrThrow($commentId);
+        if (!$this->assignmentSolutionAcl->canDeleteReviewComment($solution, $comment)) {
+            throw new ForbiddenRequestException("You are not allowed to delete this review comment");
+        }
+
+        if ($comment->getSolution()->getId() !== $id) {
+            throw new BadRequestException("Selected comment does not belong into the review of selected solution");
+        }
+    }
+
+    /**
+     * Remove one comment from a review.
+     * @DELETE
+     * @param string $id identifier of the solution
+     * @param string $commentId identifier of the review comment
+     */
+    public function actionDeleteComment(string $id, string $commentId)
+    {
+        $comment = $this->reviewComments->findOrThrow($commentId);
+        $isIssue = $comment->isIssue();
+        $this->reviewComments->remove($comment);
+
+        $solution = $this->assignmentSolutions->findOrThrow($id);
+        if ($solution->getReviewedAt() !== null) {
+            if ($isIssue) {
+                $solution->setIssuesCount($solution->getIssuesCount() - 1);
+                $this->assignmentSolutions->persist($solution);
+            }
+            $this->reviewComments->flush();
+
+            // review has been closed, report the modification
```

Comment:
Seems a bit like this comment does not belong...

Reply:
Not really, but I have made it better